docs/developer/Menu-Commands.md

```diff
@@ -0,0 +1,72 @@
+# Adding Commands to Zowe Explorer Menus
+
+This page contains developer guidance specific to adding commands in core Zowe Explorer. For more general guidance on how to add a command in VS Code, see the [VS Code Commands Extension Guide](https://code.visualstudio.com/api/extension-guides/command).
+
+## Contents
+
+- [Defining a new command](#defining-a-new-commmand)
+- [Contributing menu items](#contributing-menu-items)
+- [Menu group naming conventions](#menu-group-naming-conventions)
+
+## Defining a new commmand
+
+In order to add a command to a core Zowe Explorer menu, the command must first be defined in `packages/zowe-explorer/package.json`, under [`contributes.commands`](https://code.visualstudio.com/api/references/contribution-points#contributes.commands). The following conventions should be used when defining the respective command properties:
+
+- **`command`** property: For each Zowe Explorer view in which a command can be used (i.e. Data Sets, USS, or Jobs view), one of the prefixes in the table below should be added to the front of the `command` property to indicate what Zowe Explorer view the command applies to.
+
+  | Prefix                | Applicable view | Example usage           |
+  | --------------------- | --------------- | ----------------------- |
+  | `zowe.<command>`      | Data Sets       | `zowe.addFavorite`      |
+  | `zowe.uss.<command>`  | USS             | `zowe.uss.addFavorite`  |
+  | `zowe.jobs.<command>` | Jobs            | `zowe.jobs.addFavorite` |
+
+- **`title`** property: Be sure to localize the command's title. This can be done by following the steps for adding a new string to the `package.json` file in the [Developer's ReadMe](https://github.com/zowe/vscode-extension-for-zowe/blob/master/docs/Developer's%20ReadMe.md#adding-strings).
+
+  - **Tip:** When choosing the command's key name, consider prefixing it with `mvs.`, `uss.`, or `jobs.` to indicate if the key is for the Data Sets, USS, or Jobs view, respectively. (For example, the key `%uss.addFavorite%` is used for the `addFavorite` command in the USS view.) If the command is more related to profiles than to a specific view, consider prefixing the key name with `profile.`.
```

Comment:
Should we add a `tip` if the extender is not going to prefix their command with either mvs, uss or jobs? Like in what format or is that covered in VS Code Commands Extension Guide ?

Reply:
This guidance is for Zowe Explorer developers (rather than extenders). There is a general tip on localization key naming conventions in the linked [Developer's ReadMe](https://github.com/zowe/vscode-extension-for-zowe/blob/master/docs/Developer's%20ReadMe.md#adding-strings). However, this is more of a best practice/suggestion rather than a strict guideline, since Zowe Explorer localization key naming does not really interact with extender contributions (unlike context menu group naming).

Reply:
Ok. That makes sense. 👍

Reply:
Some additional thoughts: The majority of our strings for the Data Sets view have no prefix on them. Ideally, they would have the `ds` (or `mvs`) prefix, but again, that involves a lot of changes in the code. We could do it, but then the scope and potential impact of implementing this guideline retroactively becomes much bigger. I was thinking we could just encourage it as a best practice going forward for now. Then, if we want to standardize the older localization keys, it could be a separate effort in the future (maybe after all the conformance criteria work is more settled).

Reply:
If possible, I would suggest to open an issue and then we could visit that in V2. Is it okay if you open one?

Reply:
@jellypuno I have created issue #1345 to address this. (Clean up Zowe Explorer command localization keys).

Reply:
Thank you @lauren-li ! you are awesome! 🎉